src/fetchClient.ts

```diff
@@ -0,0 +1,131 @@
+import { ResponseType } from './responseType';
+
+import { FetchClientConfiguration, RequestOptions } from './index';
+type RequestParameters = Record<string, string | number>;
+
+class FetchClient {
+  private configuration: FetchClientConfiguration = {
+    baseUrl: '',
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    headers: { 'Content-Type': 'application/json' },
+  };
+
+  private getPath({
+    path,
+    options,
+  }: {
+    path: string;
+    params?: RequestParameters;
+    options?: RequestOptions;
+  }) {
+    const baseUrl = options?.baseUrl || this.configuration.baseUrl;
+    if (!baseUrl) {
+      throw new Error(
+        'FetchClient is not configured. Please run initApiClient() or pass a custom baseUrl.'
+      );
+    }
+    return [
+      baseUrl.endsWith('/') ? baseUrl.slice(0, -1) : baseUrl,
+      path.startsWith('/') ? path.slice(1) : path,
+    ].join('/');
+  }
+
+  private getHeaders(options?: RequestOptions): Record<string, string> {
+    return {
+      ...this.configuration.headers,
+      ...options?.headers,
+      ...(options?.accessToken
+        ? { Authorization: `Bearer ${options.accessToken}` }
+        : {}),
+    };
+  }
+
+  private static async returnData(response: Response): Promise<ResponseType> {
+    const text = await response.text();
+    const data = text.length > 0 ? JSON.parse(text) : {};
```

Comment:
wrap with try/catch and use .json()

Reply:
I think this is more performant than try/catch.